### PR TITLE
Issue#140

### LINKED ISSUE #140: Add User routes
Add `User` routes in README.md. 

Refer [this PR](https://github.com/CodeTrophs/Backend/pull/136) for details that should be included for different routes added.

### PR DESCRIPTION
**Pull request by ankitvashisht12**
Fixes #140

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -268,3 +268,130 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
             "commentId": "COMMENT_ID"
          }
          ```
+         
+##### **User routes**
+1. ```GET [PROTECTED ROUTE] /v1/user/profiles``` - This route lists all the profiles.
+      * Request body - 
+         ```json
+         {}
+         ```
```
Comment (Dishebh) [inline README.md:277]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -268,3 +268,130 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
+         {
+            "page": 1,
+            "per_page": 10
+         }
+         ```
+2. ```GET [PROTECTED ROUTE] /v1/user/profile/:id``` - This route lists profile of a particular user.
+      * Request body - 
+         ```json
+         {}
+         ```
```
Comment (Dishebh) [inline README.md:302]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -268,3 +268,130 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
+         ```json 
+         {
+            "id": "USER_ID"
+         }
+         ```
+3. ```GET [PROTECTED ROUTE] /v1/user/profile``` - This route lists ?.
+      * Request body - 
+         ```json
+         {}
+         ```
```
Comment (Dishebh) [inline README.md:324]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -268,3 +268,130 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
+          }
+         }
+         ```
+      * Status code of the response - `200`
+   
+5. ```PATCH [PROTECTED ROUTE] /v1/user/socials``` - This route lets user update their social data.
+      * Request body - 
+         ```json
+         {}
+         ```
```
Comment (Dishebh) [inline README.md:362]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -268,3 +268,130 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
+          }
+         }
+         ```
+      * Status code of the response - `200`
+
+6. ```POST [PROTECTED ROUTE] /v1/user/avatar``` - This route lets user upload their profile image .
+      * Request body - 
+         ```json
+         {}
+         ```
```
Comment (Dishebh) [inline README.md:379]:
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -268,3 +268,130 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
             "commentId": "COMMENT_ID"
          }
          ```
+         
+##### **User routes**
+1. ```GET [PROTECTED ROUTE] /v1/user/profiles``` - This route lists all the profiles.
```
Comment (Dishebh) [inline README.md:273]:
```suggestion
1. ```GET [ADMIN PROTECTED ROUTE] /v1/user/profiles``` - This route lists all the profiles.
```

File: README.md [hunk 7]
```diff
@@ -268,3 +268,130 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
+         ```
+      * Status code of the response - `200`
+      * Query parameters in request (default) - 
+         ```json
+         {
+            "page": 1,
+            "per_page": 10
+         }
+         ```
+2. ```GET [PROTECTED ROUTE] /v1/user/profile/:id``` - This route lists profile of a particular user.
```
Comment (Dishebh) [inline README.md:298]:
```suggestion
2. ```GET [ADMIN PROTECTED ROUTE] /v1/user/profile/:id``` - This route lists profile of a particular user.
```

File: README.md [hunk 8]
```diff
@@ -268,3 +268,130 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
+         }
+         ```
+      * Status code of the response - `200`
+      * Route Parameters in request URL - 
+         ```json 
+         {
+            "id": "USER_ID"
+         }
+         ```
+3. ```GET [PROTECTED ROUTE] /v1/user/profile``` - This route lists ?.
```
Comment (Dishebh) [inline README.md:320]:
```suggestion
3. ```GET [USER PROTECTED ROUTE] /v1/user/profile``` - This route lists ?.
```

File: README.md [hunk 9]
```diff
@@ -268,3 +268,130 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
+         {
+          "data": {
+            "_id": "USER_ID",
+            "name": "USER NAME",
+            "profileImage": "USER PROFILE IMAGE"
+          }
+         }
+         ```
+      * Status code of the response - `200`
+4. ```PATCH [PROTECTED ROUTE] /v1/user/profile``` - This route lets user update their profile.
```
Comment (Dishebh) [inline README.md:336]:
```suggestion
4. ```PATCH [USER PROTECTED ROUTE] /v1/user/profile``` - This route lets the user update their profile.
```

File: README.md [hunk 10]
```diff
@@ -268,3 +268,130 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
+         "data": {
+            "_id": "USER_ID",
+            "name": "USER NAME",
+            "profileImage": "USER PROFILE IMAGE"
+          }
+         }
+         ```
+      * Status code of the response - `200`
+   
+5. ```PATCH [PROTECTED ROUTE] /v1/user/socials``` - This route lets user update their social data.
```
Comment (Dishebh) [inline README.md:358]:
```suggestion
5. ```PATCH [USER PROTECTED ROUTE] /v1/user/socials``` - This route lets user update their social data.
```

File: README.md [hunk 11]
```diff
@@ -268,3 +268,130 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
+          "data": {
+            "_id": "USER_ID",
+            "name": "USER NAME",
+            "profileImage": "USER PROFILE IMAGE"
+          }
+         }
+         ```
+      * Status code of the response - `200`
+
+6. ```POST [PROTECTED ROUTE] /v1/user/avatar``` - This route lets user upload their profile image .
```
Comment (Dishebh) [inline README.md:375]:
```suggestion
6. ```POST [USER PROTECTED ROUTE] /v1/user/avatar``` - This route lets user upload their profile image .
```

File: README.md [hunk 12]
```diff
@@ -268,3 +268,130 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
+         ```
+      * Response body - 
+         ```json
+         {
+          "totalPages": "TOTAL PAGES AVAILABLE",
+          "currentPage": "CURRENT PAGE NUMBER",
+          "users": {
+            "_id": "USER_ID",
+            "name": "USER NAME",
+            "profileImage": "USER PROFILE IMAGE"
```
Comment (Dishebh) [inline README.md:286]:
```suggestion
     '_id' : "USER_ID",
    'name': ...,
    'profileImage',
    'about',
    'title',
    'role',
    'skills',
    'socials',
```

File: README.md [hunk 13]
```diff
@@ -268,3 +268,130 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
+         ```json
+         {}
+         ```
+      * Response body - 
+         ```json
+         {
+          "data": {
+            "_id": "USER_ID",
+            "name": "USER NAME",
+            "profileImage": "USER PROFILE IMAGE"
```
Comment (Dishebh) [inline README.md:309]:
```suggestion
     '_id',
    'name',
    'profileImage',
    'about',
    'title',
    'role',
    'skills',
    'socials',
```

File: README.md [hunk 14]
```diff
@@ -268,3 +268,130 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
```
Comment (Dishebh) [inline README.md:331]:
```suggestion
     '_id',
    'name',
    'profileImage',
    'about',
    'title',
    'role',
    'skills',
    'socials',
```

File: README.md [hunk 15]
```diff
@@ -268,3 +268,130 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
+            "title": "TITLE"
+         }
+         ```
+      * Response body - 
+         ```json
+         {
+         "data": {
+            "_id": "USER_ID",
+            "name": "USER NAME",
+            "profileImage": "USER PROFILE IMAGE"
```
Comment (Dishebh) [inline README.md:352]: Same thing here as well

File: README.md [hunk 16]
```diff
@@ -268,3 +268,130 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
+         ```json
+         {}
+         ```
+      * Response body - 
+         ```json
+         {
+          "data": {
+            "_id": "USER_ID",
+            "name": "USER NAME",
+            "profileImage": "USER PROFILE IMAGE"
```
Comment (Dishebh) [inline README.md:369]: here too

File: README.md [hunk 17]
```diff
@@ -268,3 +268,140 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
+          "currentPage": "CURRENT PAGE NUMBER",
+          "users": {
+            "_id": "USER_ID",
+            "name",
+            "profileImage",
+            "about",
+            "title",
+            "role",
+            "skills",
+            "socials",
```
Comment (Dishebh) [inline README.md:287]: No, I meant "name" as in "name": "USER_NAME", and so on.

File: README.md [hunk 18]
```diff
@@ -268,3 +268,140 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
+         }
+         ```
+      * Status code of the response - `200`
+      * Route Parameters in request URL - 
+         ```json 
+         {
+            "id": "USER_ID"
+         }
+         ```
+3. ```GET [USER PROTECTED ROUTE] /v1/user/profile``` - This route lists ?.
```
Comment (Dishebh) [inline README.md:322]:
```suggestion
3. ```GET [USER PROTECTED ROUTE] /v1/user/profile``` - This route lists profile of the logged-in user.
```

File: README.md [hunk 19]
```diff
@@ -268,3 +268,140 @@ Open the `codetrophs` directory in your code editor of choice and edit files und
...
+            "role",
+            "skills",
+            "socials",
+          }
+         }
+         ```
+      * Status code of the response - `200`
+   
+5. ```PATCH [USER PROTECTED ROUTE] /v1/user/socials`,`` - This route lets user update their social data.
+      * Response body -
```
Comment (Dishebh) [inline README.md:367]: No `Request body` here?

File: README.md (hunk 17 above)
Comment (ankitvashisht12) [reply README.md:287]: ohh.. okay, i'll correct it

File: README.md (hunk 19 above)
Comment (ankitvashisht12) [reply README.md:367]: sorry, i missed it somehow! Will correct it too.

Comment (Dishebh) [reply README.md:383]: The request body should have inputs like `GitHub, Twitter, Linkedin, website`. https://github.com/CodeTrophs/Backend/blob/ec527bd6c7286488f16c2dd5b5f2444cb213da9b/src/controllers/v1/user.js#L85

Let me know in the slack if you need any help.

**Review by Dishebh — APPROVED** [review]

**Pull request merged**